File: unityPart/Assets/Scripts/Lua/CSharpCallLua/CSharpCallLua.lua

local export = {
    controllersMap = {},
    managersMap = {},
}

-- export.Controllers = {}

export.managersMap.LogicManager = require("Managers.LogicManager")
export.timer = require("CSharpCallLua.Timer")
-- export.CameraController = require("Controllers.CameraController")
export.SetAvatarAndEnemy = function(self, avatar, enemy)
    Global.avatar = avatar
    Global.enemy = enemy
    export.controllersMap.CameraController.SetAvatar(avatar)
end
export.OnGameBegin = function(self)
    local GameManager = Global.Managers.GameManager
    GameManager.OnGameBegin()
end

return export

File: unityPart/Assets/Scripts/TemplateClass/CSharpCallLua.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using Base;
using XLua;

[CSharpCallLua]
public interface CSharpCallLua
{
    public Dictionary<string, IController> controllersMap { get; set; }
    public Dictionary<string, IManager> managersMap { get; set; }
    public ITimer timer { get; set; }
    public void SetAvatarAndEnemy(GameObject avatar, GameObject enemy);
    public void OnGameBegin();
    //public IController cameraController;
    //public List<IController> controllers;
}
